torch/_inductor/kernel/mm_scaled.py

```diff
@@ -279,20 +529,37 @@ def tuned_scaled_mm(
         choices.append(aten_choice)
 
     static_shape, is_nonzero = _is_static_problem(layout)
+
     if is_nonzero and use_triton_template(layout, enable_float8=True):
-        for config in scaled_mm_configs(m, n, k):
-            if k == 16 and config.kwargs["BLOCK_M"] >= 64:
-                continue  # Triton crashes in this case
-            kwargs = scaled_mm_options(
-                config, m, n, k, layout, scale_a, scale_b, use_fast_accum
-            )
-            # possibly appends a TritonTemplateCaller to choices
-            triton_template.maybe_append_choice(
-                choices,
-                input_nodes=input_nodes,
-                layout=layout,
-                **kwargs,
-            )
+        if has_triton_tma_device():
```

Comment:
IIUC, TMA has a limitation of the innermost dim being 16-bytes aligned. In Triton, the tensor should also be contiguous. Are those conditions assumed to be satisfied or do we need to add checks?

Reply:
currently assumed but will add checks

Reply:
Actually since we would just be copying all the exact checks here from here:  https://github.com/pytorch/pytorch/blob/5accae41977cc162c8c27143c9444a22d0c78ae3/torch/_meta_registrations.py#L5598-L5629to the lowering, I think okay to not have these checks. We also constrain to fx strides as well